From b8daf5a4aec0eba213cbea515d61a7663a0c040e Mon Sep 17 00:00:00 2001
From: Afzal Mohammed <afzal@ti.com>
Date: Wed, 21 Dec 2011 13:40:14 +0530
Subject: [PATCH 0411/1092] arm:omap:am33xx: LCD backlight support

Turn on LCD backlight based on board and
profile detected (i.e if LCD is present)

Signed-off-by: Afzal Mohammed <afzal@ti.com>
Signed-off-by: Vaibhav Hiremath <hvaibhav@ti.com>
Signed-off-by: Manjunathappa, Prakash <prakash.pm@ti.com>
---
 arch/arm/mach-omap2/board-am335xevm.c | 40 +++++++++++++++++++++++++++++++++++
 1 file changed, 40 insertions(+)

diff --git a/arch/arm/mach-omap2/board-am335xevm.c b/arch/arm/mach-omap2/board-am335xevm.c
index 9ceddfb..ab0441c 100644
--- a/arch/arm/mach-omap2/board-am335xevm.c
+++ b/arch/arm/mach-omap2/board-am335xevm.c
@@ -17,6 +17,7 @@
 #include <linux/i2c.h>
 #include <linux/module.h>
 #include <linux/i2c/at24.h>
+#include <linux/gpio.h>
 
 #include <mach/hardware.h>
 #include <mach/board-am335xevm.h>
@@ -194,6 +195,41 @@ static void _configure_device(int evm_id, struct evm_dev_cfg *dev_cfg,
 	}
 }
 
+/* Convert GPIO signal to GPIO pin number */
+#define GPIO_TO_PIN(bank, gpio) (32 * (bank) + (gpio))
+
+#define AM335X_LCD_BL_PIN	GPIO_TO_PIN(0, 7)
+
+/* Module pin mux for eCAP0 */
+static struct pinmux_config ecap0_pin_mux[] = {
+	{"ecap0_in_pwm0_out.gpio0_7", AM33XX_PIN_OUTPUT},
+	{NULL, 0},
+};
+
+static int backlight_enable;
+
+static void enable_ecap0(int evm_id, int profile)
+{
+	backlight_enable = true;
+}
+
+static int __init ecap0_init(void)
+{
+	int status = 0;
+
+	if (backlight_enable) {
+		setup_pin_mux(ecap0_pin_mux);
+
+		status = gpio_request(AM335X_LCD_BL_PIN, "lcd bl\n");
+		if (status < 0)
+			pr_warn("Failed to request gpio for LCD backlight\n");
+
+		gpio_direction_output(AM335X_LCD_BL_PIN, 1);
+	}
+	return status;
+}
+late_initcall(ecap0_init);
+
 /* Low-Cost EVM */
 static struct evm_dev_cfg low_cost_evm_dev_cfg[] = {
 	{NULL, 0, 0},
@@ -201,6 +237,8 @@ static struct evm_dev_cfg low_cost_evm_dev_cfg[] = {
 
 /* General Purpose EVM */
 static struct evm_dev_cfg gen_purp_evm_dev_cfg[] = {
+	{enable_ecap0,	DEV_ON_DGHTR_BRD, (PROFILE_0 | PROFILE_1 |
+						PROFILE_2 | PROFILE_7) },
 	{NULL, 0, 0},
 };
 
@@ -211,6 +249,8 @@ static struct evm_dev_cfg ind_auto_mtrl_evm_dev_cfg[] = {
 
 /* IP-Phone EVM */
 static struct evm_dev_cfg ip_phn_evm_dev_cfg[] = {
+	{enable_ecap0,	DEV_ON_DGHTR_BRD, PROFILE_NONE},
+	{lcdc_init,	DEV_ON_DGHTR_BRD, PROFILE_NONE},
 	{NULL, 0, 0},
 };
 
-- 
1.7.11.2

